utils.go

```diff
@@ -32,3 +37,47 @@ func MerkleLogger(moduleName string) zerolog.Logger {
 	logger := zerolog.New(customConsoleWriter).With().Str("module", moduleName).Timestamp().Logger()
 	return logger
 }
+
+// An MerkleRootQueue is a min-heap.
+type MerkleRootQueue []MerkleRootEntry
+
+func (h MerkleRootQueue) Len() int { return len(h) }
+func (h MerkleRootQueue) Less(i, j int) bool {
+	return h[i].BundleId < h[j].BundleId
+}
+func (h MerkleRootQueue) Swap(i, j int) { h[i], h[j] = h[j], h[i] }
+
+func (h *MerkleRootQueue) Push(x any) {
+	// Push and Pop use pointer receivers because they modify the slice's length,
+	// not just its contents.
+	*h = append(*h, x.(MerkleRootEntry))
+}
+
+func (h *MerkleRootQueue) Pop() any {
+	old := *h
+	n := len(old)
+	x := old[n-1]
+	*h = old[0 : n-1]
+	return x
+}
+
+func GetMerkleFileName(poolId int) string {
+	return fmt.Sprintf("merkle_roots_pool_%v", poolId)
+}
+
+func GetPoolHeights(pools []Pool) map[int]int {
```

Comment:
Would rename to something like `GetPoolWrittenMerkleRootCount` as height is often used as synonym for pool key.

Reply:
Or `GetCurrentBundleCountByPool`